### PR TITLE
Refactor domain model

### PR DESCRIPTION
**Pull request by robertgmoss**

### CODE REVIEW

**Comment by duncangrant** [comment]
What is it that everyone seems to have against interfaces?

**Comment by robertgmoss** [comment]
I like interfaces, generally, but in this case it provides no benefit since there is only ever going to be one implementation

**Comment by duncangrant** [comment]
Please rebase against master

**Comment by duncangrant** [comment]
@sjcorbett  only just added the ambari domain models.  Do you know what his intentions were for them?  As they weren't n use I'm happy for them to be removed and he can revert the commit when he needs them.

File: ambari/src/main/java/io/brooklyn/ambari/rest/Mapable.java [hunk 1]
```diff
@@ -18,12 +18,26 @@
...
+import java.util.List;
 import java.util.Map;
 
+import com.google.common.collect.ImmutableList;
+
 /**
  * Created by duncangrant on 22/01/15.
  */
-public interface AsMap {
+public interface Mapable {
```
Comment (sjcorbett) [inline ambari/src/main/java/io/brooklyn/ambari/rest/Mapable.java]: Two ps in mappable.

File: pom.xml [hunk 2]
```diff
@@ -89,6 +89,21 @@
             </dependency>
         </dependencies>
     </dependencyManagement>
+    
+    <dependencies>
```
Comment (sjcorbett) [inline pom.xml:93]: Why add these dependencies to the parent pom?

**Comment by sjcorbett** [comment]
@duncangrant The intention for the Ambari domain objects was to use them rather than bother writing our own - there's no need for us to write a `RecommendationResponse`, Ambari have written and annotated it with `@JsonParam` annotations for us.

**Comment by robertgmoss** [comment]
@sjcorbett Unfortunately these weren't sufficient - the json sent back could not be mapped into them properly.  Was easier in the end to clean up the existing model.

File: pom.xml (hunk 2 above)
Comment (robertgmoss) [inline pom.xml:93]: My Eclipse has errors all over without them.

File: ambari/src/main/java/io/brooklyn/ambari/rest/Mapable.java (hunk 1 above)
Comment (robertgmoss) [inline ambari/src/main/java/io/brooklyn/ambari/rest/Mapable.java]: Good spot

**Comment by duncangrant** [comment]
I'm happy for this to be merged assuming @sjcorbett agrees

File: pom.xml (hunk 2 above)
Comment (sjcorbett) [inline pom.xml:93]: I'm inclined to blame your Eclipse. I guess we can include them for now but would be good if you worked out the proper configuration/why Eclipse is complaining.

**Comment by sjcorbett** [comment]
Suggest dropping 7f43b87 from this pull request and squashing the remaining commits to one or two logical commits. @robertgmoss I can do this if you like, otherwise will leave to you.

**Comment by robertgmoss** [comment]
thanks @sjcorbett please do

**Comment by duncangrant** [comment]
Closing as merged by PR #14

**Pull request closed without merging**